crates/types/src/blockchain/header.rs

```diff
@@ -365,17 +369,31 @@ impl PartialBlockHeader {
         self,
         transactions: &[Transaction],
         message_ids: &[MessageId],
+        events: &[Event],
```

Comment:
This is the wrong event type, these events are internal to fuel and not the same as the relayer events. This should be based on the events that are used in the `process_da` function from the executor, ie. 

https://github.com/FuelLabs/fuel-core/blob/master/crates/types/src/services/relayer.rs#L11

Reply:
this is resolved now but theres some cleanup to do still.